client/src/app/services/order.service.ts

```diff
@@ -48,32 +48,31 @@ export class OrderService {
     this.cartSubject.next([]);
   }
 
+  //* Get full cart
   getAllItems() {
     return this.httpClient.get<OrderResponse>(this.baseUrlOrder);
   }
 
+  getOrder(id) {
+    return this.httpClient.get(`${this.baseUrlOrder}/invoice/${id}`);
+  }
+
   getTotalCarts() {
-    return this.httpClient.get(`${this.baseUrl}/sum`);
+    return this.httpClient.get(`${this.endpoint}/sum`);
   }
 
   completeOrder(body) {
     return this.httpClient
       .post(`${this.baseUrlOrder}/finish`, body)
       .subscribe((res: any) => {
-        if (res.message === 'success') {
-          this.clear();
-          this.openInvoiceDialog(res.data.order._id);
-          this.router.navigate(['products']);
-        } else {
-          // send error
-        }
+        //TODO: handle error
+        if (res.message !== 'success') return;
+        this.clear();
+        this.openInvoiceDialog(res.data.order._id);
```

Comment:
is it ok to open a dialog and navigate at the same time?

Reply:
the dialog is a modal, if you click out, it closes

Reply:
and the routed page is in the background ?

Reply:
yep